Lib/test/test_clinic.py

```diff
@@ -2863,6 +2877,121 @@ def test_cloned_func_with_converter_exception_message(self):
                 func = getattr(ac_tester, name)
                 self.assertEqual(func(), name)
 
+    def test_depr_star_pos0_len1(self):
+        fn = ac_tester.depr_star_pos0_len1
+        self.check_depr_star(
+            fname="depr_star_pos0_len1", pnames="'a'",
+            good_calls=[
+                lambda: fn(a=None),
+            ],
+            bad_calls=[
+                lambda: fn(None),
+            ])
+
+    def test_depr_star_pos0_len2(self):
+        fn = ac_tester.depr_star_pos0_len2
+        self.check_depr_star(
+            fname="depr_star_pos0_len2", pnames="'a' and 'b'",
+            good_calls=[
+                lambda: fn(a=None, b=None),
+            ],
+            bad_calls=[
+                lambda: fn(None, b=None),
+                lambda: fn(None, None),
+            ])
+
+    def test_depr_star_pos0_len3_with_kwd(self):
+        fn = ac_tester.depr_star_pos0_len3_with_kwd
+        self.check_depr_star(
+            fname="depr_star_pos0_len3_with_kwd", pnames="'a', 'b' and 'c'",
+            good_calls=[
+                lambda: fn(a=None, b=None, c=None, d=None),
+            ],
+            bad_calls=[
+                lambda: fn(None, b=None, c=None, d=None),
+                lambda: fn(None, None, c=None, d=None),
+                lambda: fn(None, None, None, d=None),
+            ])
```

Comment:
Another readability nit: I find all the `None`s everywhere distracting, just because of the number of characters it takes up on the line... what if we passed in `0` instead of `None`? It would be more concise.

Reply:
Or 1, 2, 3,...